src/legacy/ui/public/visualize/loader/pipeline_helpers/utilities.ts

```diff
@@ -140,30 +141,39 @@ export const getFormat: FormatFactory = mapping => {
           if (val === '__missing__') {
             return params.missingBucketLabel;
           }
-          const parsedUrl = {
-            origin: window.location.origin,
-            pathname: window.location.pathname,
-            basePath: npStart.core.http.basePath,
+
+          const options = {
+            parsedUrl: {
+              origin: window.location.origin,
+              pathname: window.location.pathname,
+              basePath: npStart.core.http.basePath,
+            },
           };
           // @ts-ignore
-          return format.convert(val, undefined, undefined, parsedUrl);
+          return format.convert(val, options);
         };
       },
       convert: (val: string, type: string) => {
         const format = getFieldFormat(params.id, mapping.params);
         if (val === '__other__') {
           return params.otherBucketLabel;
         }
+
         if (val === '__missing__') {
           return params.missingBucketLabel;
         }
-        const parsedUrl = {
-          origin: window.location.origin,
-          pathname: window.location.pathname,
-          basePath: npStart.core.http.basePath,
+
+        const options = {
+          type,
+          parsedUrl: {
+            origin: window.location.origin,
+            pathname: window.location.pathname,
+            basePath: npStart.core.http.basePath,
+          },
         };
+
         // @ts-ignore
-        return format.convert(val, type, undefined, parsedUrl);
+        return format.convert(val, options);
```

Comment:
I don't like an idea to move type into options
I see interface for `convert` method as: 
`format.convert(value, type, options)`

Reply:
Thanks. Fixed